sql3/planner/inbuiltfunctionsstring.go

```diff
@@ -644,3 +668,65 @@ func (n *callPlanExpression) EvaluateReplicate(currentRow []interface{}) (interf
 	return strings.Repeat(stringArg, int(intArg)), nil
 
 }
+
+// function to get the index of substring in the given string
+func (n *callPlanExpression) EvaluateCharIndex(currentRow []interface{}) (interface{}, error) {
+
+	argEval, err := n.args[0].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+	if argEval == nil {
+		return nil, nil
+	}
+	subString, ok := argEval.(string)
+	if !ok {
+		return nil, sql3.NewErrInternalf("unexpected type converion %T", argEval)
```

Comment:
fb-1817 adds a new function that wraps around ErrInternalF. Rebase once that is merged.

Reply:
Rebased and updated